### PR TITLE
V2.8.0 additional gps channels

### PR DESCRIPTION
**Pull request by brentpicasso**
Adding two new GPS channels:  Altitude and quality indicators.

Also, have GPS module's update rate provisioned at a rate that matches the loggerConfig update rate.

### CODE REVIEW

File: src/gps/gpsTask.c [hunk 1]
```diff
@@ -34,7 +35,8 @@ void setGpsDataLogging(bool enable) {
 
 void GPSTask(void *pvParameters) {
 	Serial *serial = get_serial(SERIAL_GPS);
-	int rc = GPS_device_provision(serial);
+	uint8_t targetSampleRate = decodeSampleRate(getWorkingLoggerConfig()->GPSConfigs.speed.sampleRate);
+	int rc = GPS_device_provision(targetSampleRate, serial);
 	if (!rc){
```
Comment (Jeff-Ciesielski) [inline src/gps/gpsTask.c]: Camel case

File: stm32_base/hal/gps_skytraq/gps_device_skytraq.c [hunk 2]
```diff
@@ -526,7 +533,23 @@ static gps_cmd_result_t configureUpdateRate(GpsMessage *gpsMsg, Serial *serial,
...
+    }
+    else if (sampleRate > 5){
+        return 10;
+    }
+    else if (sampleRate > 1){
+        return 5;
+    }
+    return 1;
+}
+
```
Comment (Jeff-Ciesielski) [inline stm32_base/hal/gps_skytraq/gps_device_skytraq.c]: Ditto

**Comment by Jeff-Ciesielski** [comment]
:+1:   Looks fine to me.

**Comment by brentpicasso** [comment]
Thanks. I feel like I'm forgetting something - is our standard to use
underscores instead of camelCase? The whole firmware codebase is pretty
much camelcase..

On Sat, Mar 28, 2015 at 10:02 PM, Jeff Ciesielski <notifications@github.com>
wrote:

> [image: :+1:] Looks fine to me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/27#issuecomment-87358787>
> .
>



-- 
Brent Picasso
*Autosport Labs <http://www.autosportlabs.com>*
*Technology for Race and Street*

**Comment by Jeff-Ciesielski** [comment]
The standard is to use underscores.

**Pull request merged**